### ISSUE TITLE
level_probs generation code

### ISSUE THREAD

**Issue by wzczc** [zhiyuanyou/DeQA-Score#2]
Thank you for your work! After reading the paper, I tried to generate the level_probs myself, but they are quite different from the ones provided in your dataset mete files. Could you please provide the code for level_probs generation?

**Comment by zhiyuanyou** [comment]
Thanks for your interests. 

We need some time to organize these codes. I think it will be OK to release in this weekend.

**Comment by wzczc** [comment]
> Thanks for your interests.
> 
> We need some time to organize these codes. I think it will be OK to release in this weekend.

Thanks for your time! Looking forward to it!

**Comment by zhiyuanyou** [comment]
Please see **Soft Label Construction** in README for details.

**Comment by wzczc** [comment]
Thanks a lot! One more question, I notice you use mos_norm and std_norm to generate soft labels, if I only train with one dataset, is it the same to use mos and std to generate labels?

**Comment by zhiyuanyou** [comment]
No. The mos must be normalized to [1, 5], which is what the discretization process needs. And the std should be normalized with mos.

**Comment by wzczc** [comment]
Got it, thank you very much!

**Comment by dongdk** [comment]
hi @zhiyuanyou what if the dataset does not have std information, how to create the gt label and train the model?

**Comment by zhiyuanyou** [comment]
If without the std information, we can: 

- see Equation (A2) in Appendix, define a soft label with only two level tokens. 

<img width="555" alt="Image" src="https://github.com/user-attachments/assets/dcc26739-23eb-4a76-93fd-1c3819c749b4" />

- see Sec. B.1 in Appendix, estimate an std.

**Comment by dongdk** [comment]
get it.
another question: why use original mos and std to compute fedelity loss instead of those normed, and why use scorer func to get the predictions，which forward again？

**Comment by dongdk** [comment]
by the way, why not use pairwise ranking loss, like this one https://github.com/bo-zhang-cs/CGS-Pytorch/blob/main/croppingModel.py#L129?

**Comment by zhiyuanyou** [comment]
> get it. another question: why use original mos and std to compute fedelity loss instead of those normed, and why use scorer func to get the predictions，which forward again？

1. The ground-truth P(A>B) results of (1) the original mos and std and (2) the normed mos and std are the **same**. 
2. I did not fully get your point.

**Comment by zhiyuanyou** [comment]
> by the way, why not use pairwise ranking loss, like this one https://github.com/bo-zhang-cs/CGS-Pytorch/blob/main/croppingModel.py#L129?

We adopt the fidelity loss since this loss has been verified by several previous IQA methods. Maybe you mentioned pairwise ranking loss also works. We will note this and try it in our future experiments. Thanks!
